cli/src/cli_util.rs

```diff
@@ -1606,10 +1633,11 @@ See https://martinvonz.github.io/jj/latest/working-copy/#stale-working-copy \
             print_failed_git_export(ui, &refs)?;
         }
 
-        self.user_repo = ReadonlyUserRepo::new(tx.commit(description));
```

Comment:
Perhaps, we'll also need to disable git exports because the operation is discarded. (There's another `git::export_refs()` caller in `snapshot_working_copy()`.)

It might make sense to split `may_update_working_copy`/`working_copy_shared_with_git` flags into `may_snapshot`/`may_update`/`may_import`/`may_export`, but I'm not sure if that works out.